### ISSUE TITLE
Bug in training loss computation?

### ISSUE THREAD

**Issue by mattrmd** [chemcognition-lab/chemixhub#5]
It appears that there is a bug in the training and validation losses (I have only shown the training loss here):
https://github.com/chemcognition-lab/chemixhub/blob/356c16b99e38b56a91b5219d33db410fc372d538/src/mixhub/model/train.py#L58-L79
I think the following code should be used instead:
```python
    for i, batch in enumerate(train_loader):
        train_indices = batch["ids"].to(device)
        train_features = batch["features"].to(device)
        train_fractions = batch["fractions"].to(device)
        train_context = batch["context"].to(device)
        train_labels = batch["label"].to(device)

        optimizer.zero_grad()

        y_pred = model(train_features, train_indices, train_fractions, train_context)
        loss = loss_fn(y_pred.view(-1), train_labels.view(-1))
        metrics.update(y_pred.flatten(), train_labels)

        loss.backward()
        optimizer.step()

        # should not be i, otherwise it will count n*(n-1)/2 batches instead of n
        num_batch += 1
        train_loss += loss.detach().cpu().item()

    # We should be dividing by n. I think the n+1 was previously necessary because of the single-batch
    # case where the bug would set num_batch = 0 since the first i = 0.
    overall_train_loss = train_loss / num_batch
    overall_train_metrics = compute_metrics(metrics, {"loss": overall_train_loss})
```
I have not checked whether this bug(?) shows up in other files.
